bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxProperty.java

```diff
@@ -0,0 +1,113 @@
+package org.openhab.binding.caddx.internal;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+@NonNullByDefault
+public class CaddxProperty {
+    // private
+    String name;
+    String type; // 'Int', 'String', 'Bit'
+    int byteFrom;
+    int byteLength;
+    int bitFrom;
+    int bitLength;
+    boolean external;
+    String id;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    // Constructor
+    public CaddxProperty(String id, int byteFrom, int byteLength, int bitFrom, int bitLength, String type, String name,
+            boolean external) {
+        this.id = id;
+        this.name = name;
+        this.type = type;
+        this.byteFrom = byteFrom;
+        this.byteLength = byteLength;
+        this.bitFrom = bitFrom;
+        this.bitLength = bitLength;
+        this.external = external;
+    }
+
+    public String getValue(byte[] message) {
+        int mask;
+        int val;
+
+        if (type == "Int") {
+            if (bitFrom == 0 && bitLength == 0) {
+                mask = 255;
+                val = message[byteFrom - 1] & mask;
+            } else {
+                mask = ((1 << ((bitLength - bitFrom))) - 1) << bitFrom;
+                val = (message[byteFrom - 1] & mask) >> bitFrom;
+            }
+
+            return Integer.toString(val);
+        }
+
+        if (type == "String") {
+            byte[] str = Arrays.copyOfRange(message, byteFrom - 1, byteFrom + byteLength);
+            return new String(str);
+        }
+
+        if (type == "Bit") {
+            return (((message[byteFrom - 1] & (1 << bitFrom)) > 0) ? "true" : "false");
+        }
+
+        throw new IllegalArgumentException("type [" + type + "] is unknown.");
+    }
+
+    public String toString(byte[] message) {
+        int mask;
+        int val;
+
+        if (type == "Int") {
```

Comment:
don't use `==` with String instances.
```suggestion
        if ("Int".equals(type)) {
```
Please make appropriate changes to where it may apply.

Reply:
Oooops. Corrected.